core/server/src/main/java/alluxio/worker/DefaultAlluxioWorker.java

```diff
@@ -338,11 +342,20 @@ public MetricsSystem getWorkerMetricsSystem() {
   public void waitForReady() {
     while (true) {
       if (mThriftServer.isServing()
-          && WorkerIdRegistry.getWorkerId() != WorkerIdRegistry.INVALID_WORKER_ID
+          && mWorkerId.get() != null
```

Comment:
Would it make sense to use wait/notify instead of a sleep loop? Though the sleep pattern is much more common in the code base.